app/assets/javascripts/search.js

```diff
@@ -1,31 +1,145 @@
 $(document).ready(function() {
 
-  /* open the Adv Search UI immediately if param set */
+  /*************************************************************************************************/
+  /* ** Search Form ** */
+
+  var advOptionsVisible = function() {
+    return $("fieldset.adv").is(":visible");
+  };
+
+  var FADE_SPEED = 200;
+
+  // manage the basic search input button via the search-terms field
+  var searchTerms = $(".m-search-ui .search-terms");
+  var buttonToggler = function() {
+    if (searchTerms && searchTerms.val() && searchTerms.val().length == 0) {
+      $("#search-button").hide();
+      $(".m-search-ui .input-group-addon.magnifier").fadeIn(FADE_SPEED);
+    }
+    else {
+      if (!advOptionsVisible()) {
+        $("#adv-options").fadeIn(FADE_SPEED);
+      }
+      $(".m-search-ui .input-group-addon.magnifier").hide();
+      $("#search-button").fadeIn(FADE_SPEED);
+    }
+  };
+
+  var showAdvOptions = function() {
+    $("fieldset.adv").fadeIn(FADE_SPEED);
```

Comment:
rather than referencing these selectors in various functions, should we assign them to a var so we can easily update our js if markup changes? Also, thoughts on adding selectors with `js` prefix so we know they are there for js?